src/spec.ts

```diff
@@ -319,9 +319,42 @@ export function alwaysNoOcclusion(spec: ExtendedUnitSpec): boolean {
   return vlEncoding.isAggregate(spec.encoding);
 }
 
-export function fieldDefs(spec: ExtendedUnitSpec): FieldDef[] {
+function hasFieldDef(fieldDefs: FieldDef[], fieldDef: FieldDef): boolean {
+  for (let i = 0; i < fieldDefs.length; i++) {
+    if (identical(fieldDefs[i], fieldDef)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function fieldDefsOfAllLayers(spec: LayerSpec): FieldDef[] {
```

Comment:
I don't like this method name and believe that with more efficient implementation, we don't really need this method.